ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -97,21 +105,32 @@ type EraCrypto era = Crypto era
 -- and Mary) this type is set to SL.WitnessSet. This will eventually change,
 -- most likely with Alonzo, thus this equivalence will no longer be valid.
 class ( SL.ShelleyBasedEra era
+
       , State (LC.EraRule "PPUP" era) ~ SL.PPUPState era
       , Default (State (LC.EraRule "PPUP" era))
+
```

Comment:
I reorganized this so that it's grouped by the type being constrained (`EraRule`, `PParams`, `PParamsDelta`, `TxInBlock`, etc etc)